scripts/build.sh

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+script_name="${BASH_SOURCE:-$0}"
+script_path=$(realpath "$script_name")
+scripts_dir_path=$(dirname "$script_path")
+
+# shellcheck disable=SC1091
+source "${scripts_dir_path}/lib/_functions.sh"
+# shellcheck disable=SC1091
+source "${scripts_dir_path}/lib/_logger.sh"
+
+function usage {
+    cat <<EOF
+This script performs the following tasks in this specific order:
+1. Generates a list of Operator manifests for a SonataFlow project
+2. Builds the workflow image using 
+3. Optionally deploys.
+    3.1 Pushes the image to a registry (specidied in the image path)
+    3.2 Applies the generated manifests in the current k8s namespace
+
+Usage: 
+    $script_name [flags]
+
+Flags:
+    -i|--image string                 The image path to use for the workflow (required).
+    -b|--builder-image string         Overrides the image to use for building the workflow image.
+    -h|--help                         Prints this help message.
+    -r|--runtime-image string         Overrides the image to use for running the workflow.
+    -w|--workflow-directory string    Path to the directory containing the workflow's files (the 'src' directory). Default: current directory.
+    -P|--no-persistence               Skips adding persistence configuration to the sonataflow CR.
+       --apply                        Applies the generated manifests in the current namespace.
+       --push                         Pushes the image to the registry after building
+
+Notes: 
+    1. The manifests will be in 'manifests' beside 'src'.
```

Comment:
Suggestion: I would put the manifests folder in a tmp directory to avoid polluting the code directory

Reply:
IMO users can ignore the folder through .gitignore (this is what I did in my repo).
BTW, if it is not uncommon we can make it configurable with a flag like `--manifests-output`. WDYT?

Reply:
yes, let's go with the flag!

Reply:
+1 for making it configurable

Reply:
ack